templates/rust/http/src/main.rs

```diff
@@ -0,0 +1,52 @@
+use actix_web::{http::Method, web, App, HttpRequest, HttpResponse, HttpServer};
+
+use env_logger as elog;
+use log::info;
+
+async fn index(req: HttpRequest) -> HttpResponse {
```

Comment:
I'm curious why this function is async. I apologize for my Rust ignorance. How is this async and how does that work in an HTTP request?

Reply:
I'm no expert, but rust async is inspired by erlang async. In rust, you have to bring your own runtime to schedule the lightweight processes defined by that async keyword. Actix uses the tokio runtime, which ultimately gets created by that "macro attribute" on line 15: `#[actix_web::main]`.

I'm not familiar with the actix-web internals but I expect at some point, something is invoking `await` on the `Future` returned by the `index` function. I know it says it's returning an `HttpResponse`, but because that `async` keyword is there, it's really returning a `Future`.

Were you looking for more detail than that?

Reply:
I think so.